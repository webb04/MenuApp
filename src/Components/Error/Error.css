.error-list {
  background-color: rgba(218,55,67,.2);
  margin: 20px 20px 50px 20px;
  border-radius: 3px;
  padding: 10px 20px;
}

.error {
  color: #DA3743;
  text-align: left;
}

.empty {
  background-color: transparent;
  padding: 0;
  height: 72px;
}

@media screen and (max-width: 768px) {
  .error-list {
    float: left;
    width: 90%;
    margin: 20px 5% 50px 5%;
    box-sizing: border-box;
  }
}
