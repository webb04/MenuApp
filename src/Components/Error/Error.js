import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Error.css';

class Error extends Component {
  render() {
    const { errors } = this.props;
        return errors.length > 0 ?
          <div className="error-list">
            {
              errors.map((error, key) =>
                <p className="error" key={ `error-${key}` }>
                  { error }
                </p>
              )
            }
          </div>
        : <div className="error-list empty"></div>
      }
}

Error.defaultProps = {
  errors: []
}

Error.propTypes = {
  errors: PropTypes.array
}

export default Error;
