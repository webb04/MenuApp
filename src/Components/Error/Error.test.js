import React from 'react';
import ReactDOM from 'react-dom';
import Error from './Error';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('Does not render container if no errors', () => {
  const wrapper = shallow(<Error errors={ [] } />);
  expect(wrapper.find('.empty')).toHaveLength(1);
});

it('Renders correct error messages', () => {
  const errors = [
    'Pierre strongly advises against the prawn cocktail and salmon fillet combination',
    'Each diner cannot have more than one of the same course.',
    'Each person must have at least two courses, one of which must be a main.'
  ];
  const wrapper = shallow(<Error errors={ errors } />);
  expect(wrapper.find('.empty')).toHaveLength(0);
  expect(wrapper.find('.error')).toHaveLength(3);

  const error = renderer
    .create(<Error errors={ errors } />)
    .toJSON();
  expect(error).toMatchSnapshot();

});
