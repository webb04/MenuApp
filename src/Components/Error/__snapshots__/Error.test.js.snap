// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correct error messages 1`] = `
<div
  className="error-list"
>
  <p
    className="error"
  >
    Pierre strongly advises against the prawn cocktail and salmon fillet combination
  </p>
  <p
    className="error"
  >
    Each diner cannot have more than one of the same course.
  </p>
  <p
    className="error"
  >
    Each person must have at least two courses, one of which must be a main.
  </p>
</div>
`;
